aten/src/ATen/native/cuda/Math.cuh

```diff
@@ -31,7 +31,7 @@ static inline __host__ __device__ scalar_t calc_digamma(scalar_t in) {
     return static_cast<scalar_t>(INFINITY);
   }
 
-  bool x_is_integer = x == ::floor(x);
+  bool x_is_integer = x == std::floor(x);
```

Comment:
I'm a little confused. Does gcc compile this, or does nvcc do it? The code ultimately gets run on device, right?

Reply:
The function is defined as `__host__ __device__`, that means it will be compiled twice

Reply:
@zou3519 The main point of this kind of change in this PR is that, the CUDA doc is silent on this issue (given that `std::floor` exists), and assuming `::floor` is the same as `std::floor` is inconsistent with standard C++. I think it would best if we can be more compliant to avoid a **potential** issue, which by itself at least doesn't hurt.

Reply:
Okay, to make sure I understand what's going on: where is `floor` in `::floor` coming from?

Reply:
It can come from a C definition (as declared in a C header file, which is solely defined for double). The standard C++ library is silent on `::floor`, while it defines `std::floor`. Does this answer your question?

Reply:
That does, thanks for the clarification